### DISCUSSION TITLE
SCP connection with smartcard?

### DISCUSSION THREAD

**Discussion by LucyDemooon** [veeso/termscp#214] [Q&A]
Hello,

is there a way to get smartcards working with termscp?
my ssh keys are all stored on smartcards.

regards lucy

**Comment by veeso** [comment]
Hi Lucy, unfortunately, it's not. I've never worked with smartcards and there is very poor support in rust for it, so I don't even think it's achievable

**Reply by LucyDemooon** [reply]
I have this 2 lines in my shell profile:
```
export SSH_AUTH_SOCK=$(gpgconf --list-dirs agent-ssh-socket)
gpgconf --launch gpg-agent 
```

> export SSH_AUTH_SOCK=$(gpgconf --list-dirs agent-ssh-socket): This command sets the environment variable SSH_AUTH_SOCK. This variable is typically used to specify the path to the SSH agent socket, which is used for communication between SSH requests. In this case, gpgconf --list-dirs agent-ssh-socket is used to determine the path to the SSH agent socket used by the GPG agent, and this path is then exported into SSH_AUTH_SOCK.

> gpgconf --launch gpg-agent: This command launches the GPG agent, a component of the GPG system used for managing private keys and creating signatures. In this specific scenario, gpg-agent is started with the --launch flag to ensure it is running.


Possibly one might be able to access the export variable when connecting without entering a password?
